packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -216,85 +217,72 @@ export class SelectFileName
+    try {
+      packageDirs = await SfdxPackageDirectories.getPackageDirectoryPaths();
+    } catch (e) {
       if (
-        !this.explorerDir &&
-        this.globDirs(rootPath, this.globKeyWord).length === 0
+        e.name !== 'NoPackageDirectoryPathsFound' &&
+        e.name !== 'NoPackageDirectoriesFound'
       ) {
-        notificationService.showErrorMessage(
-          nls.localize(
-            'parameter_directory_strict_not_available',
-            this.globKeyWord
-          )
-        );
-        return { type: 'CANCEL' };
+        throw e;
       }
-      outputdir = this.explorerDir
-        ? path.relative(rootPath, this.explorerDir)
-        : await vscode.window.showQuickPick(
-            this.globDirs(rootPath, this.globKeyWord),
-            {
-              placeHolder: nls.localize('parameter_gatherer_enter_dir_name')
-            } as vscode.QuickPickOptions
-          );
     }
+    let dirOptions = this.getDefaultOptions(packageDirs);
+    let outputdir = await this.showMenu(dirOptions);
+
+    if (outputdir === this.customDirOption) {
+      dirOptions = this.getCustomOptions(getRootWorkspacePath());
+      outputdir = await this.showMenu(dirOptions);
+    }
+
     return outputdir
       ? { type: 'CONTINUE', data: { outputdir } }
       : { type: 'CANCEL' };
   }
-}
-export class SelectPrioritizedDirPath extends SelectDirPath {
-  public globDirs(srcPath: string, priorityKeyword?: string): string[] {
-    const unprioritizedRelDirs = new glob.GlobSync(
-      path.join(srcPath, '**/')
-    ).found.map(value => {
-      let relativePath = path.relative(srcPath, path.join(value, '/'));
-      relativePath = path.join(relativePath, '');
-      return relativePath;
-    });
-    if (priorityKeyword) {
-      const notPrioritized: string[] = [];
-      const prioritized = unprioritizedRelDirs.filter(dir => {
-        if (dir.includes(priorityKeyword)) {
-          return true;
-        } else {
-          notPrioritized.push(dir);
-        }
-      });
-      return prioritized.concat(notPrioritized);
-    }
-    return unprioritizedRelDirs;
+
+  public getDefaultOptions(packageDirectories: string[]): string[] {
+    const options = packageDirectories.map(packageDir =>
+      path.join(packageDir, this.defaultOutput, this.typeDir)
+    );
+    options.push(this.customDirOption);
+    return options;
   }
-}
 
-export class SelectStrictDirPath extends SelectDirPath {
-  public globDirs(srcPath: string, priorityKeyword?: string): string[] {
-    const globPattern = priorityKeyword
-      ? path.join(srcPath, '**/', priorityKeyword + '/')
-      : path.join(srcPath, '**/');
-    const relativeDirs = new glob.GlobSync(globPattern).found.map(value => {
-      let relativePath = path.relative(srcPath, path.join(value, '/'));
-      relativePath = path.join(relativePath, '');
+  public getCustomOptions(rootPath: string): string[] {
+    return new glob.GlobSync(path.join(rootPath, '**/')).found.map(value => {
```

Comment:
I think this is going to break on Windows, can you verify ?

Reply:
@lcampos do we need a path utility in our core library maybe? This seems to be a common problem. Maybe if we prevented the use of `require('path`)` and used our own utility, it would cut down on these windows bugs?

Reply:
No, using `path` is the right way to address this since its a Nodejs module and it does a good job across platforms.